x/staking/keeper/msg_server.go

```diff
@@ -278,6 +263,20 @@ func (k msgServer) BeginRedelegate(goCtx context.Context, msg *types.MsgBeginRed
 	if err != nil {
 		return nil, err
 	}
+
+	delegation, found := k.GetDelegation(ctx, delegatorAddress, valSrcAddr)
+	if !found {
+		return nil, status.Errorf(
+			codes.NotFound,
+			"delegation with delegator %s not found for validator %s",
+			msg.DelegatorAddress, msg.ValidatorSrcAddress,
+		)
+	}
+
+	if delegation.Exempt {
+		return nil, types.ErrRedelegationNotAllowedForExemptDelegation
```

Comment:
Why not have the same behavior as with `Undelegate`, i.e., check if the (exempt_shares - undelegated shares) * exemption_factor >= total_tokeniz_shares?

Reply:
I think you are correct

Reply:
Probably, it should be doing like undelegation.